charts/karpenter/README.md.gotmpl

```diff
@@ -15,7 +15,7 @@ You can follow the detailed installation instruction in the [documentation](http
 helm upgrade --install --namespace karpenter --create-namespace \
   karpenter oci://public.ecr.aws/karpenter/{{ template "chart.name" . }} \
   --version v{{ template "chart.version" . }} \
-  --set serviceAccount.annotations.eks\.amazonaws\.com/role-arn=${KARPENTER_IAM_ROLE_ARN} \
+  --set "serviceAccount.annotations.eks\.amazonaws\.com/role-arn=${KARPENTER_IAM_ROLE_ARN}" \
```

Comment:
Do we need this? I think this has always worked for me. 

If anything, would this work?
```suggestion
  --set-string serviceAccount.annotations.eks\.amazonaws\.com/role-arn=${KARPENTER_IAM_ROLE_ARN} \
```

Reply:
Looks like this was updated here: #5025. I was trying to understand the case where this fails and couldn't come up with anything good. I don't want to revert it per-se since we merged the PR, but I'd like to understand when this fails with `--set` in general

Reply:
I think we can try to hear back from the original contributor but I don't want to block this PR on this

Reply:
From what I can understand, I'm assuming that it has to do with [globbing](https://www.shellcheck.net/wiki/SC2086) of the variable if it's not put into a string from the bash side.

Reply:
See #5061